File: problems/136_Single_Number.c
#include <stdio.h>

int singleNumber(int* nums, int numsSize) {
    int uniqueN = 0, index = 0;
    for (index =0; index < numsSize; index++) {
        uniqueN ^= nums[index];
        printf("%d ", uniqueN);
    }
    printf("\n");

    return uniqueN;
}

int main () {
    int ret = 0;
    int nums_1[] = {2, 2, 1};
    int nums_2[] = {4, 1, 2, 1, 2};
#if 0
    ret = singleNumber(nums_1, sizeof(nums_1)/sizeof(int));
    printf("1.Answer of 1st-Q: %d \n", ret);
#endif 
    ret = singleNumber(nums_2, sizeof(nums_2)/sizeof(int));
    printf("2.Answer of 2nd-Q: %d \n", ret);

    return 0;   
}

File: problems/136_Single_Number.py
from collections import Counter
""" My Solution.
class Solution(object):
    def singleNumber(self, nums):
        num_ct = Counter(nums)
        for key, cnt in num_ct.items():
            if cnt == 1
                return key
"""

#Another Solution.
class Solution(object):
    def singleNumber(self, nums):
        uniqueN = 0
        for val in nums:
            uniqueN ^= val

        return uniqueN

def main():
    nums_1 = [2,2,1]
    nums_2 = [4,1,2,1,2]
    
    leet = Solution()
    print(f"1.Answer of 1st-Q: {leet.singleNumber(nums_1)}")
    print(f"2.Answer of 2nd-Q: {leet.singleNumber(nums_2)}")

if __name__ == "__main__":
    main()